hibernate-reactive-core/src/test/java/org/hibernate/reactive/schema/GlobalTemporaryTableTest.java

```diff
@@ -0,0 +1,125 @@
+/* Hibernate, Relational Persistence for Idiomatic Java
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * Copyright: Red Hat Inc. and Hibernate Authors
+ */
+package org.hibernate.reactive.schema;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.concurrent.CompletionStage;
+
+import org.hibernate.boot.registry.StandardServiceRegistryBuilder;
+import org.hibernate.cfg.Configuration;
+import org.hibernate.dialect.temptable.TemporaryTable;
+import org.hibernate.reactive.BaseReactiveTest;
+import org.hibernate.reactive.testing.SqlStatementTracker;
+
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Test;
+
+import io.vertx.junit5.Timeout;
+import io.vertx.junit5.VertxTestContext;
+import jakarta.persistence.Column;
+import jakarta.persistence.Entity;
+import jakarta.persistence.Id;
+import jakarta.persistence.SecondaryTable;
+import jakarta.persistence.Table;
+
+import static java.util.concurrent.TimeUnit.MINUTES;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.hibernate.query.sqm.mutation.internal.temptable.GlobalTemporaryTableStrategy.CREATE_ID_TABLES;
+import static org.hibernate.reactive.util.impl.CompletionStages.voidFuture;
+
+@Timeout(value = 10, timeUnit = MINUTES)
+public class GlobalTemporaryTableTest extends BaseReactiveTest {
+
+	private static SqlStatementTracker sqlTracker;
+
+	@Override
+	protected Collection<Class<?>> annotatedEntities() {
+		return List.of( Bar.class );
+	}
+
+
+	@Override
+	public CompletionStage<Void> deleteEntities(Class<?>... entities) {
+		// Skip delete step.
+		// We don't insert any value, so there's nothing to delete
+		// Avoid having extra stuff in the log that's not relevant to the test
+		return voidFuture();
+	}
+
+	@Override
+	protected void setProperties(Configuration configuration) {
+		super.setProperties( configuration );
+		configuration.setProperty( CREATE_ID_TABLES, "false" );
```

Comment:
I think you also need to test that when the property is set to `true`, it creates the temporary tables as expected (on the databases that supports it).

You can see an example of how we test different session factory configurations in [SchemaUpdateTest](https://github.com/hibernate/hibernate-reactive/blob/42d5c0479d6bab0a9a49a360f3999fe82d39328a/hibernate-reactive-core/src/test/java/org/hibernate/reactive/schema/SchemaUpdateTest.java#L44)

Reply:
While reviewing the code, I noticed we also have a property `hibernate.query.mutation_strategy.global_temporary.drop_tables`, could we also test this, please?
See `GlobalTemporaryTableStrategy.DROP_ID_TABLES`